### PR TITLE
Issue 1538 - hzn compatibility check needs to consider node type and …

### PR DESCRIPTION
**Pull request by linggao**
…service type

### CODE REVIEW

File: cli/hzn.go [hunk 1]
```diff
@@ -368,15 +368,18 @@ Environment Variables:
 	deploycheckLong := deploycheckCmd.Flag("long", msgPrinter.Sprintf("Show policies and userinput used for the compatibility checking.")).Short('l').Bool()
 	policyCompCmd := deploycheckCmd.Command("policy", msgPrinter.Sprintf("Check policy compatibility."))
 	policyCompNodeArch := policyCompCmd.Flag("arch", msgPrinter.Sprintf("The architecture of the node. It is required when -n is not specified. If omitted, the service of all the architectures referenced in the deployment policy will be checked for compatibility.")).Short('a').String()
+	policyCompNodeType := policyCompCmd.Flag("node-type", msgPrinter.Sprintf("The node type. The valid values are 'device' and 'cluster'. It is required when -n is not specified. If omitted when -n is not specified, 'device' node type will be assumed.")).Short('t').String()
```
Comment (dabooz) [inline cli/hzn.go:371]: This statement contradicts itself...Is -t required or not? I would add device as the default (with the .Default() function call as is done elsewhere in hzn cli) to this line of code and adjust the help text to match.

File: cli/hzn.go [hunk 2]
```diff
@@ -368,15 +368,18 @@ Environment Variables:
...
 	policyCompDepPolId := policyCompCmd.Flag("deployment-pol-id", msgPrinter.Sprintf("The Horizon exchange deployment policy ID. Mutually exclusive with -B. If you don't prepend it with the organization id, it will automatically be prepended with the node's organization id.")).Short('b').String()
 	policyCompBPolFile := policyCompCmd.Flag("business-pol", "").Hidden().String()
 	policyCompDepPolFile := policyCompCmd.Flag("deployment-pol", msgPrinter.Sprintf("The JSON input file name containing the Deployment policy. Mutually exclusive with -b.")).Short('B').String()
 	policyCompSPolFile := policyCompCmd.Flag("service-pol", msgPrinter.Sprintf("(optional) The JSON input file name containing the service policy. If omitted, the service policy will be retrieved from the exchange for the service defined in the deployment policy.")).String()
+	policyCompSvcFile := policyCompCmd.Flag("service", msgPrinter.Sprintf("(optional) The JSON input file name containing the service definition. Mutually exclusive with -b. If omitted, the service defined in the deployment policy from the exchange. This flag can be repeated to specify different versions of the service.")).Strings()
```
Comment (dabooz) [inline cli/hzn.go:379]: typo in the text, change "the service defined in the deployment policy from the exchange" to "the service referenced in the deployment policy is retrieved from the exchange."

File: cli/hzn.go [hunk 3]
```diff
@@ -368,15 +368,18 @@ Environment Variables:
...
 	userinputCompCmd := deploycheckCmd.Command("userinput", msgPrinter.Sprintf("Check user input compatibility."))
 	userinputCompNodeArch := userinputCompCmd.Flag("arch", msgPrinter.Sprintf("The architecture of the node. It is required when -n is not specified. If omitted, the service of all the architectures referenced in the deployment policy or pattern will be checked for compatibility.")).Short('a').String()
+	userinputCompNodeType := userinputCompCmd.Flag("node-type", msgPrinter.Sprintf("The node type. The valid values are 'device' and 'cluster'. It is required when -n is not specified. If omitted when -n is not specified, 'device' node type will be assumed.")).Short('t').String()
```
Comment (dabooz) [inline cli/hzn.go:382]: Same comment as above for policyCompNodeType.

File: cli/hzn.go [hunk 4]
```diff
@@ -388,6 +391,7 @@ Environment Variables:
 	userinputCompPatternFile := userinputCompCmd.Flag("pattern", msgPrinter.Sprintf("The JSON input file name containing the pattern. Mutually exclusive with -p, -b and -B.")).Short('P').String()
 	allCompCmd := deploycheckCmd.Command("all", msgPrinter.Sprintf("Check all compatibilities for a deployment."))
 	allCompNodeArch := allCompCmd.Flag("arch", msgPrinter.Sprintf("The architecture of the node. It is required when -n is not specified. If omitted, the service of all the architectures referenced in the deployment policy or pattern will be checked for compatibility.")).Short('a').String()
+	allCompNodeType := allCompCmd.Flag("node-type", msgPrinter.Sprintf("The node type. The valid values are 'device' and 'cluster'. It is required when -n is not specified. If omitted when -n is not specified, 'device' node type will be assumed.")).Short('t').String()
```
Comment (dabooz) [inline cli/hzn.go:394]: Same comment as above for policyCompNodeType.

File: cli/hzn.go (hunk 1 above)
Comment (linggao) [reply cli/hzn.go:371]: Fixed

File: cli/hzn.go (hunk 3 above)
Comment (linggao) [reply cli/hzn.go:382]: Fixed

File: cli/hzn.go (hunk 4 above)
Comment (linggao) [reply cli/hzn.go:394]: Fixed

**Pull request merged**